modules/madsenseBidAdapter.js

```diff
@@ -0,0 +1,263 @@
+import {
+  logError,
+  logWarn,
+  logMessage,
+  deepSetValue,
+  mergeDeep,
+} from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER, VIDEO } from '../src/mediaTypes.js';
+import { ortbConverter } from '../libraries/ortbConverter/converter.js';
+import { config } from '../src/config.js';
+
+const BIDDER_CODE = 'madsense';
+const DEFAULT_BID_TTL = 55;
+const DEFAULT_NET_REVENUE = true;
+const DEFAULT_CURRENCY = 'USD';
+const MS_EXCHANGE_BASE_URL = 'https://ads.madsense.io/pbjs';
+
+const buildImpWithDefaults = (buildImp, bidRequest, context) => {
+  const imp = buildImp(bidRequest, context);
+
+  imp.bidfloor = imp.bidfloor || bidRequest.params.bidfloor || 0;
+  imp.bidfloorcur = imp.bidfloorcur || bidRequest.params.currency || DEFAULT_CURRENCY;
+
+  return imp;
+};
+
+const enrichRequestWithConsent = (req, bidderRequest) => {
+  if (bidderRequest.gdprConsent) {
+    deepSetValue(req, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    deepSetValue(req, 'regs.ext.gdpr', bidderRequest.gdprConsent.gdprApplies ? 1 : 0);
+  }
+
+  if (bidderRequest.uspConsent) {
+    deepSetValue(req, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+  }
+};
+
+const determineMediaType = (bid) => {
+  if (bid.mtype === 2) {
+    return VIDEO;
+  }
+
+  if (hasVideoMediaType(bid)) {
+    return VIDEO;
```

Comment:
This is where I'm still mixed up, if you won't bid on outstream video, and an impression has both video and banner media type, aren't you just going to ignore all multiformat units instead of taking them as banner?